pkg/netpol/eval/ingressanalyzer/ingress_analyzer.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2022
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ingressanalyzer
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	ocroutev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/eval"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/eval/internal/k8s"
+	"github.com/np-guard/netpol-analyzer/pkg/netpol/scan"
+)
+
+type (
+	IngressAnalyzer struct {
+		pe *eval.PolicyEngine // a struct type that includes the podsMap and
+		// some functionality on pods and namespaces which is required for ingress analyzing
+		servicesMap map[string]map[string]*k8s.Service // map from namespace to map from service name to its object
+		routesMap   map[string]map[string]*k8s.Route   // map from namespace to map from route name to its object
+		// k8sIngressMap map[string]map[string]*k8s.Ingress // map from namespace to map from ingress name to its object
+	}
+)
+
+// NewIngressAnalyzer returns a new IngressAnalyzer with an empty initial state
+func NewIngressAnalyzer() *IngressAnalyzer {
+	return &IngressAnalyzer{
+		pe:          eval.NewPolicyEngine(),
+		servicesMap: make(map[string]map[string]*k8s.Service),
+		routesMap:   make(map[string]map[string]*k8s.Route),
+	}
+}
+
+// NewIngressAnalyzerWithObjects returns a new IngressAnalyzer with relevant objects
+func NewIngressAnalyzerWithObjects(objects []scan.K8sObject) (*IngressAnalyzer, error) {
+	ia := NewIngressAnalyzer()
+	var err error
+	ia.pe, err = eval.NewPolicyEngineWithObjects(objects)
+	if err != nil {
```

Comment:
do we need to create a new policy engine, or assume it was created and just receive it as input arg? 
I think the connlist will have to create both a policy engine and an IngressAnalyzer that can use the already created policy-engine object. (it will also have to add the ingress-controller pod to get policy-based connectivity between this pod to existing pods in the cluster).

Reply:
I started with the approach sending the `PolicyEngine pointer` as a parameter to the `NewIngressAnalyzerWithObjects`,
but changed to this before committing, thinking that we may build only the `IngressAnalyzer ` explicitly in the connlist (the policyEngine will be built anyway - since both Constructors use the same objects list) - and then we can refer to the "ia.pe" to compute/ use the policyEngine.

However, I agree it is more readable to build both explicitly (committed a change)